Dfe.Academies.External.Web.UnitTest/Controllers/SchoolControllerTests.cs

```diff
@@ -3,59 +3,84 @@
 using Microsoft.AspNetCore.Mvc;
 using Microsoft.Extensions.Logging;
 using Moq;
+using Moq.Protected;
 using NUnit.Framework;
+using System;
+using System.IO;
 using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
 using System.Threading.Tasks;
 
 namespace Dfe.Academies.External.Web.UnitTest.Controllers;
 
 [Parallelizable(ParallelScope.All)]
 internal sealed class SchoolControllerTests
 {
-    /// <summary>
-    /// TODO MR:- re-test after PR55 - plumb in trams API
-    /// </summary>
-    //[Test]
-    public async Task SchoolController___Search___ReturnsResult()
+    [Test]
+    public async Task Seach___ResultsFound___ResultsReturned()
     {
         // arrange
         var mockLogger = new Mock<ILogger<SchoolController>>();
         var mockReferenceDataRetrievalService = new Mock<IReferenceDataRetrievalService>();
         var mockConversionApplicationRetrievalService = new Mock<IConversionApplicationRetrievalService>();
         var schoolController = new SchoolController(mockLogger.Object, mockReferenceDataRetrievalService.Object, mockConversionApplicationRetrievalService.Object);
         string schoolName = "wise";
+        string fullFilePath = @$"{AppDomain.CurrentDomain.BaseDirectory}ExampleJsonResponses/schoolSearchResponse.json";
+        string expectedJson = await File.ReadAllTextAsync(fullFilePath);
+        var mockFactory = SetupMockHttpClientFactory(HttpStatusCode.OK, expectedJson);
+
+        // TODO MR:- need to throw the mockFactory into the DI pipeline so that  ResilientRequestProvider consumes it
 
         // act
         var result = await schoolController.Search(schoolName);
 
-        //var viewModel = (result as ViewResult).Model;
-
         // assert
         Assert.That(result, Is.Not.Null);
         Assert.That(result.Count(), Is.Zero);
-        //Assert.That(viewModel, Is.Not.Null);
     }
 
-    /// <summary>
-    /// TODO MR:- re-test after PR55 - plumb in trams API
-    /// </summary>
-    //[Test]
-    public async Task SchoolController___ReturnSchoolDetailsPartialViewPopulated___ReturnsPartialView()
+    [Test]
+    public async Task ReturnSchoolDetailsPartialViewPopulated___ValidSchool___ReturnsPartialView()
```

Comment:
result is a 500 error in this test

Reply:
plumbed in mock IHttpClientFactory which has fixed this